packages/react-magma-dom/src/components/NativeSelect/NativeSelect.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+
+import { inputBaseStyles } from '../InputBase';
+import { InputProps } from '../Input';
+import { InputMessage } from '../Input/InputMessage';
+import { Label } from '../Label';
+import { SelectTriggerButton } from '../Select/SelectTriggerButton';
+
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { useIsInverse } from '../../inverse';
+import { I18nContext } from '../../i18n';
+import { useGenerateId } from '../../utils';
+
+/**
+ * @children required
+ */
+export interface NativeSelectProps extends InputProps {
+  disabled?: boolean;
+  hasError?: boolean;
+  helperMessage?: React.ReactNode;
+  testId?: string;
+  isInverse?: boolean;
+  optionLabel?: string;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
+}
+const StyledNativeInput = styled.select`
+  ${inputBaseStyles};
+  background: inherit;
+`;
+
+const StyledNativeSelectWrapper = styled.div<{
+  isInverse: boolean;
+}>``;
+
+export const NativeSelect = React.forwardRef<HTMLDivElement, NativeSelectProps>(
+  (props, ref) => {
+    const {
+      children,
+      containerStyle,
+      disabled,
+      errorMessage,
+      id: defaultId,
+      helperMessage,
+      isInverse: isInverseProp,
+      labelText,
+      messageStyle,
+    } = props;
+    const theme = React.useContext(ThemeContext);
+    const isInverse = useIsInverse(isInverseProp);
+    const i18n = React.useContext(I18nContext);
+    const id = useGenerateId(defaultId);
+
+    return (
+      <StyledNativeSelectWrapper
+        data-testid={props.testId}
+        isInverse={isInverse}
+        ref={ref}
+        style={containerStyle}
+        theme={theme}
+      >
+        <Label htmlFor={id} isInverse={isInverse}>
+          {labelText}
+        </Label>
+
+        <SelectTriggerButton toggleButtonProps={''} disabled={disabled}>
```

Comment:
The inverse version isn't getting the correct focus style, and the red border isn't showing up for the error message.  I think you might need to add `isInverse={isInverse}` and `hasError={!!errorMessage}` (or something similar) here.